### PR TITLE
Feat: Context menu

### PR DESCRIPTION
**Pull request by Deep-Codes**
Based on #71 

github

### CODE REVIEW

File: src/views/desktop/Desktop.tsx [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 import { useTheme } from '__/hooks';
 import css from './Desktop.module.scss';
+import ContextMenu from '__/components/Desktop/ContextMenu/ContextMenu';
 
 const DarkBackground = '/assets/wallpapers/3-1.jpg';
 const LightBackground = '/assets/wallpapers/3-2.jpg';
 
 export const Desktop = () => {
   const [theme] = useTheme();
+  const outerRef: React.LegacyRef<HTMLElement> | undefined = useRef(null);
```
Comment (PuruVJ) [inline src/views/desktop/Desktop.tsx:15]: Hi, I see you typed the useRef manually. Is there a reason not to use `useRef<HTMLDivElement>()` and this instead?

File: src/data/menu/context.menu.config.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export const contextMenuConfig = {
```
Comment (PuruVJ) [inline src/data/menu/context.menu.config.ts:1]: Hmm. Not sure I agree with keeping it as menu config. Can't think of anything right, so I'll merge it and re-architecht later after some thinking :)

File: src/views/desktop/Desktop.tsx [hunk 3]
```diff
@@ -21,7 +23,8 @@ export const Desktop = () => {
     <>
       <StartupChime />
 
-      <main className={css.main}>
+      <main ref={outerRef} className={css.main}>
+        <ContextMenu outerRef={outerRef} />
```
Comment (PuruVJ) [inline src/views/desktop/Desktop.tsx:27]: This is some clever wizardry dude!! great work here!

File: src/hooks/use-context-menu.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { useEffect, useCallback, useState } from 'preact/hooks';
+
+const useContextMenu = (outerRef: any) => {
+  const [xPos, setXPos] = useState('0px');
+  const [yPos, setYPos] = useState('0px');
+  const [menu, showMenu] = useState(false);
```
Comment (PuruVJ) [inline src/hooks/use-context-menu.ts:6]: I think this pair should be renamed as `[isMenuVisible, setIsMenuVisible]`, owing to the useState conventions

**Review by PuruVJ — CHANGES_REQUESTED** [review]
Looks great to me. Please review the comments I made.

File: src/views/desktop/Desktop.tsx (hunk 1 above)
Comment (Deep-Codes) [reply src/views/desktop/Desktop.tsx:15]: Nah no reason for that it's just that I have been using that, switched to `useRef<HTMLDivElement>()`

**Review by PuruVJ — APPROVED** [review]

**Pull request merged**